base/dates/types.jl

```diff
@@ -1,6 +1,6 @@
 # This file is a part of Julia. License is MIT: http://julialang.org/license
 
-abstract AbstractTime
+abstract AbstractTime <: AbstractScalar
```

Comment:
Hmmm... I guess this is ok. AbstractTime was kind of a catch all for all the Dates types, but this should be ok. The piece that may not make as much is the parsing Slot types, but those probably need to be reconfigured anyway.